react/features/calendar-sync/web/microsoftCalendar.js

```diff
@@ -308,6 +309,66 @@ export const microsoftCalendarApi = {
                 }));
             });
         };
+    },
+
+    /**
+     * Updates calendar event by generating new invite URL and editing the event
+     * adding some descriptive text and location.
+     *
+     * @param {string} id - The event id.
+     * @param {string} calendarId - The id of the calendar to use.
+     * @param {string} location - The location to save to the event.
+     * @returns {function(Dispatch<*>): Promise<string|never>}
+     */
+    updateCalendarEvent(id: string, calendarId: string, location: string) {
+        return (dispatch: Dispatch<*>, getState: Function): Promise<*> => {
+            const state = getState()['features/calendar-sync'] || {};
+            const token = state.msAuthState && state.msAuthState.accessToken;
+
+            if (!token) {
+                return Promise.reject('Not authorized, please sign in!');
+            }
+
+            const { dialInNumbersUrl } = getState()['features/base/config'];
+            const text = getShareInfoText(
+                location, dialInNumbersUrl !== undefined, true/* use html */);
+
+
+            const client = Client.init({
+                authProvider: done => done(null, token)
+            });
+
+            return client
+                .api(`/me/events/${id}`)
+                .get()
+                .then(response => {
+                    return {
```

Comment:
This intermediate step seems unnecessary.  If you remove this .then block, I think no other code changes.